neo/Network/P2P/LocalNode.cs

```diff
@@ -112,6 +112,37 @@ private static IEnumerable<IPEndPoint> GetIPEndPointsFromSeedList(int seedsToTak
             }
         }
 
+        public bool CheckDuplicateNonce(IActorRef remoteActor, RemoteNode remoteNode)
+        {
+            var version = remoteNode.Version;
+            var remote = remoteNode.Remote;
+
+            if (version.Nonce == Nonce)
+            {
+                LocalAddresses.Add(remote.Address);
```

Comment:
What happen if i send with the nonce of the `localAddress`? I think that every time you `Add` to a hashset, you need a clean logic, otherwise it could be possible to throw an `OutOfMemory`